benchexec/tools/crux.py

```diff
@@ -0,0 +1,86 @@
+
+    def version(self, executable):
+        s = self._version_from_tool(executable)
+        return s[s.find("version:"):]
+
+    def determine_result(self, run):
+        override_pat = re.compile("No implementation or override found for pointer: \"(.+?)\"")
+
+        for line in run.output:
+            # There are still a good number of functions for which Crux lacks
+            # overrides (see, for example, issue #187). Rather than reporting
+            # FALSIFIED for such programs (which will dock us points), we will
+            # instead conservatively return UNKNOWN, which doesn't lose (or
+            # gain) points. To make it more obvious which programs are UNKNOWN
+            # due to failing overrides versus incomplete goals, we include the
+            # name of the failing override in parentheses after UNKNOWN, which
+            # will show up in benchexec's reports.
+            m = override_pat.search(line)
+            if m:
+                return result.RESULT_UNKNOWN + "(no override: " + m.group(1) + ")"
+            # Crucible does not currently support inline assembly
+            elif "unsupported LLVM value: ValAsm" in line:
+                return result.RESULT_UNKNOWN + "(inline assembly)"
+            # Crucible does not currently support translating `long double`s
+            # (https://github.com/GaloisInc/crucible/issues/810)
+            elif "unsupported LLVM value: ValFP80" in line:
+                return result.RESULT_UNKNOWN + "(long double)"
+            elif "Verification result: VERIFIED" in line:
+                return result.RESULT_TRUE_PROP
+            elif "Verification result: FALSIFIED (valid-free)" in line:
+                return result.RESULT_FALSE_FREE
+            elif "Verification result: FALSIFIED (valid-deref)" in line:
+                return result.RESULT_FALSE_DEREF
+            elif "Verification result: FALSIFIED (valid-memtrack)" in line:
+                return result.RESULT_FALSE_MEMTRACK
+            elif "Verification result: FALSIFIED (valid-memcleanup)" in line:
+                return result.RESULT_FALSE_MEMCLEANUP
+            elif "Verification result: FALSIFIED (no-overflow)" in line:
+                return result.RESULT_FALSE_OVERFLOW
+            elif "Verification result: FALSIFIED (termination)" in line:
+                return result.RESULT_FALSE_TERMINATION
+            elif "Verification result: FALSIFIED (unreach-call)" in line:
+                return result.RESULT_FALSE_REACH
+            elif "Verification result: FALSIFIED" in line:
+                return result.RESULT_FALSE_PROP
+            elif "Verification result: UNKNOWN" in line:
+                return result.RESULT_UNKNOWN + "(incomplete)"
+            elif "Verification result: ERROR" in line:
+                return result.RESULT_ERROR
+        return result.RESULT_UNKNOWN
+
+    def program_files(self, executable):
+        return [executable] + self.REQUIRED_PATHS
```

Comment:
As is, this method doesn't really make sense, because `self.REQUIRED_PATHS` is empty. So, first `REQUIRED_PATHS` should be set. If everything in the tool directory is expected to be relevant for execution, then `["."]` can be used as value, otherwise simply give a list of relevant files and directories.

But then, the current implementation of `program_files` does not work, because it needs to return paths that are relevant to the current working directory (and paths in `REQUIRED_PATHS` of course are not, because constants cannot know what the current working directory is). But the default implementation of `program_files` already handles this by assuming that the paths in `REQUIRED_PATHS` are relative to the path of `executable`. So just delete this method.

Reply:
Hm, I don't recall why exactly I opted to override this. I suppose I was paranoid since the Crux bindist includes a separate `bin/` directory where the executables live, so I wanted to be absolutely sure the files under `bin/` were picked up. But as you've pointed out, I didn't even implement `REQUIRED_PATHS` correctly, so I don't think this paranoia was necessary. Everything still works after deleting `program_files`, so I went with that approach.

Reply:
It works locally, but for SV-COMP `REQUIRED_PATHS` is necessary. So please add this with either `["."]` or a list of paths, depending on what fits better.

Reply:
I'm OK with everything every file (both at the top level and under `bin`) being included, so I've added `REQUIRED_PATHS = ["."]`.